lektor/admin/static/js/widgets/ToastEditor.jsx

```diff
@@ -0,0 +1,189 @@
+    const currentEditor = editor.getCurrentModeEditor()
+    let editorHeight
+    try {
+      // markdown
+      const editorEl = currentEditor.editorContainerEl.getElementsByClassName('CodeMirror-sizer')[0]
+      editorHeight = window.$(editorEl).outerHeight(true)
+    } catch (e) {
+      // wysiwyg
+
+      // get height of all children in editor
+      const editorEl = currentEditor.$editorContainerEl[0].firstChild
+      let editorChildren = Array.from(editorEl.children)
+      const editorChildrenHeights = editorChildren.map(el => window.$(el).outerHeight(true))
+      editorHeight = editorChildrenHeights.reduce((a, b) => a + b)
+
+      // add on padding/border/margin of editor element
+      const marginTop = parseInt(window.getComputedStyle(editorEl).marginTop)
+      const marginBottom = parseInt(window.getComputedStyle(editorEl).marginBottom)
+      const paddingTop = parseInt(window.getComputedStyle(editorEl).paddingTop)
+      const paddingBottom = parseInt(window.getComputedStyle(editorEl).paddingBottom)
+      const borderTop = parseInt(window.getComputedStyle(editorEl).borderTopWidth)
+      const borderBottom = parseInt(window.getComputedStyle(editorEl).borderBottomWidth)
+      editorHeight += (marginTop + marginBottom + paddingTop + paddingBottom + borderTop + borderBottom)
+    }
+
+    const totalHeight = editorHeight + 31 + 31
+    const totalHeightStr = totalHeight + 'px'
+    editor.height(totalHeightStr)
+  }
+
+  onChange () {
+    if (!this.state.editor) {
+      return
+    }
+
+    const markdown = this.state.editor.getMarkdown()
+
+    // recalculate height - delay is required to get most up to date height
+    setTimeout(function () {
+      this.recalculateHeight(this.state.editor)
+    }.bind(this), 10)
+
+    // send markdown up
+    const contentChanged = markdown !== this.content
+    if (this.props.onChange && this.content !== null && contentChanged) {
+      this.content = markdown
+      this.props.onChange(markdown)
+    }
+  }
+
+  // focus and blur are used to implement the border-blur found on other widgets
+  focus () {
+    this.element.classList.add('active')
+  }
+  blur () {
+    this.element.classList.remove('active')
+  }
+
+  componentDidMount () {
+    const editor = new Editor({
+      el: this.element,
+      initialValue: this.props.value,
+      initialEditType: (this.props.type.default_view === 'richtext') ? 'wysiwyg' : 'markdown',
+      useCommandShortcut: true,
+      usageStatistics: false,
+      previewStyle: null,
+      hideModeSwitch: true,
+      toolbarItems: toolbarItems,
+      hooks: {
+        addImageBlobHook: this.onImageUpload
+      },
+      events: {
+        load: this.onLoad,
+        change: this.onChange,
+        stateChange: this.onChange,
+        contentChangedFromWysiwyg: () => { setTimeout(this.onChange, 100) },
+        focus: this.focus,
+        blur: this.blur
+      },
+      exts: ['lektorImage', 'lektorLink']
+    })
+    this.setState({
```

Comment:
Since this never changes I don't think it should be part of the state, I think it can be be `this.editor` just like `this.element`.